base/shell/explorer/syspager.cpp

```diff
@@ -178,6 +178,73 @@ class CNotifyToolbar :
     void Initialize(HWND hWndParent, CBalloonQueue * queue);
 };
 
+
+class CHardErrorThread
+{
+    LONG m_bThreadRunning;
+    DWORD m_Status;
+    DWORD m_dwResponseType;
+    CStringW m_Title;
+    CStringW m_Text;
+public:
+
+    CHardErrorThread():
+        m_bThreadRunning(FALSE),
+        m_Status(NULL),
+        m_dwResponseType(NULL)
+    {
+    }
+
+    VOID ThreadProc()
+    {
+        HRESULT hr;
+        CComPtr<IUserNotification> pnotification;
+
+        OleInitialize(NULL);
+
+        hr = CoCreateInstance(CLSID_UserNotification,
+                              NULL,
+                              CLSCTX_INPROC_SERVER,
+                              IID_PPV_ARG(IUserNotification, &pnotification));
+        if (FAILED_UNEXPECTEDLY(hr))
+            return;
+
+        hr = pnotification->SetBalloonInfo(m_Title, m_Text, NIIF_WARNING);
```

Comment:
You should certainly also add some info tooltip text: iirc. when you have these sorts of notifications on Windows, you can hover also on the notification icon to get a tooltip.